packages/orbit-components/src/Separator/index.tsx

```diff
@@ -1,77 +1,79 @@
 "use client";
 
 import * as React from "react";
-import styled, { css } from "styled-components";
+import cx from "clsx";
 
-import type * as Common from "../common/types";
-import type { Theme } from "../defaultTheme";
-import defaultTheme from "../defaultTheme";
-import getSpacingToken from "../common/getSpacingToken";
-import { left, right } from "../utils/rtl";
-import type { Props, Indent, Align } from "./types";
-import useTheme from "../hooks/useTheme";
+import type { Props, Align } from "./types";
+import { getSpaceAfterClasses } from "../common/tailwind";
 
-function capitalize(string: string) {
-  return string.charAt(0).toUpperCase() + string.slice(1);
+enum BorderTypes {
+  none = "border-none",
+  solid = "border-solid",
+  dashed = "border-dashed",
+  dotted = "border-dotted",
+  double = "border-double",
 }
 
-function getIndentAmount({
-  theme,
-  $indent,
-}: // StyledContainer takes more props, so this shouldn't be exact
-{
-  theme: Theme;
-  $indent: Indent;
-}) {
-  return $indent === "none" ? 0 : theme.orbit[`space${capitalize($indent)}`];
+enum Indent {
+  none = "none",
+  small = "small",
+  medium = "medium",
+  large = "large",
+  XLarge = "XLarge",
+  XXLarge = "XXLarge",
 }
 
-const StyledContainer = styled.div<{ $align: Align; $indent: Indent }>`
-  ${({ $align }) => css`
-    box-sizing: border-box;
-    width: 100%;
-    padding-${right}: ${$align === "left" && getIndentAmount};
-    padding-${left}: ${$align === "right" && getIndentAmount};
-    padding: ${$align === "center" && `0 ${getIndentAmount}`};
-  `};
-`;
+function getIndentAmount(indent: keyof typeof Indent, align: Align) {
```

Comment:
While reviewing, I noticed that this is very non-intuitive… If we say `align="left" indent="large"` we suddenly have a big padding… on the right side. That is the exact opposite of what's expected. An indentation refers to a spacing at the start, and if it's aligned to the left, the indentation space should be a padding on the left.

This is a problem that already existed, so it's not introduced by this PR. I'd say it could be done later, but we're introducing a breaking change already so we could avoid two BC in a row… Also, to fix it is not so costly… so I'd consider doing it. We either rename the prop (I can't propose a good name, tbh) or we switch the classes applied, which can be more confusing for migrating, but will actually be semantically correct.

Reply:
I would say it's rather a problem of naming than a problem of behavior. It should work like that if we say `align="left"`, the padding should be on the right side and vice versa, so after the Separator actually. So I think it's correct in terms of behavior. Maybe we just need to rename it, so it's more descriptive for behavior 🤔